src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -599,6 +676,90 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             }
         }
 
+        // This fills in the arguments and parameters arrays in evaluation order.
+        private void BuildArgumentsInEvaluationOrder(
+            SyntaxNode syntax,
+            Symbol methodOrIndexer,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<BoundExpression> arguments,
+            /* out */ ArrayBuilder<BoundExpression> argumentsBuilder,
+            /* out */ ArrayBuilder<ArgumentKind> argumentKindsBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> parametersBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> missingParametersBuilder)
+        {
+            ImmutableArray<ParameterSymbol> parameters = methodOrIndexer.GetParameters();
+            PooledHashSet<int> processedPeremeters = PooledHashSet<int>.GetInstance();
+
+            for (int a = 0; a < arguments.Length; ++a)
+            {
+                BoundExpression argument = arguments[a];
+
+                int p = (!argsToParamsOpt.IsDefault) ? argsToParamsOpt[a] : a;
+                var parameter = parameters[p];
+
+                bool processed = processedPeremeters.Add(p);
+                Debug.Assert(processed);
+
+                bool isParamArray = false;
+                if (expanded && p == parameters.Length - 1)
+                {
+                    isParamArray = true;
+
+                    ArrayBuilder<BoundExpression> paramArray = ArrayBuilder<BoundExpression>.GetInstance();
+                    paramArray.Add(argument);
+
+                    int remainingArgument = a + 1;
+                    for (; remainingArgument < arguments.Length; ++remainingArgument)
+                    {
+                        int remainingParameter = (!argsToParamsOpt.IsDefault) ? argsToParamsOpt[remainingArgument] : remainingArgument;
+                        if (remainingParameter != parameters.Length - 1)
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            paramArray.Add(arguments[remainingArgument]);
+                        }
+                    }
+                    // Set loop variable to index of the last processed argument.
+                    a = remainingArgument - 1;
+
+                    var paramArrayType = parameters[p].Type;
+                    var arrayArgs = paramArray.ToImmutableAndFree();
+
+                    argument = CreateParamArrayArgument(syntax, methodOrIndexer, paramArrayType, arrayArgs);
+                }
+
+                argumentsBuilder.Add(argument);
+                argumentKindsBuilder.Add(isParamArray ? ArgumentKind.ParamArray : ArgumentKind.Explicit);
+                parametersBuilder.Add(parameter);
+            }
+
+            // Create an empty array for omitted param array argument.
```

Comment:
We do not yet provide any guaranteed order of default arguments, but I think it'd be expected to list them in parameter order (after all explicated argument in evaluation order), so this need to be moved to after all the other default arguments got processed, so an empty param array appears at the end of list.